src/common/service/compute_service_attributes.ts

```diff
@@ -0,0 +1,16 @@
+import { HomeAssistant, ServiceAttribute } from "../../types";
+
+export const computeServiceAttributes = (
```

Comment:
We should use this also in the automation editor and services dev panel

Reply:
Would be best if we would make 1 element that does all the work that we can use on all places we allow to set a service call.

Reply:
Just the table or the service picker and table combined?
Problem with combining is that the format is different between Lovelace and automation/service dev. Lovelace has service_data while the others have data

Reply:
We could align that? And otherwise, pass it as a property/attribute?